Teapot/RequestParameter.swift

```diff
@@ -50,13 +56,6 @@ public enum RequestParameter {
     }
 
     public init(_ data: Data) {
-        let json = try? JSONSerialization.jsonObject(with: data, options: [])
-        if let array = json as? Array<Dictionary<String, Any>> {
```

Comment:
Alright, so what was going on is that data would only be stored as data if it's not able to convert it to an array or a dictionary?

And now it will be stored as data. And if we try to access an array or a dictionary it will return the data as such if it's possible to covert it to that.

Reply:
That is correct - basically, we were automatically decoding JSON data, storing it as an array or dictionary, then re-encoding it. Since the `JSONSerialization` decode/recode methods are different from the `Codable` stuff, the data sent to the server was not identical to the data used to sign the request.